OpenTelemetry (S3M) Host and OS apps (#1878)
SumoLogic/sumologic-documentation: docs/integrations/hosts-operating-systems/opentelemetry/macos-opentelemetry.md

The Sumo Logic App for macOS allows you to monitor the performance and resource utilization of hosts and processes that your mission critical applications are dependent upon. The app consists of predefined searches and dashboards that provide visibility into your environment for real-time or historical analysis.

Preconfigured dashboards provide insight into CPU load, memory, network, file descriptors, page faults, and TCP connectors. This app uses OpenTelemetry, an open-source collector for metrics.

We use the OpenTelemetry collector for metrics collection.

OpenTelemetry collector runs on the macOS machine, and uses the [Host Metrics Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver) to obtain host and process metrics, and the [Sumo Logic OpenTelemetry Exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/sumologicexporter) to send the metrics to Sumo Logic.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Mac-OpenTelemetry/Mac-Schematics.png')} alt="Schematics" />

## Fields Created in Sumo Logic for macOS

> Following are the tags which will be created as part of macOS App install, if not already present. 

Reviewer: ```suggestion
Following are the [fields](https://help.sumologic.com/docs/manage/fields/) which will be created as part of macOS App install, if not already present. 
```

> **Across apps**

Reviewer: ```suggestion
```

- **`sumo.datasource`** - Has fixed value of **mac**.

## Collecting Logs, Metrics & installing macOS app

Here are the steps for collecting metrics and installing the app.

### Step 1: Set up Collector

> If you want to use an existing Otel Collector then this step can be skipped by selecting the option of using an existing Collector.

Reviewer: ```suggestion
:::note
If you want to use an existing OpenTelemetry Collector, you can skip this step by selecting the "Use an existing Collector" option.
:::

To create a new Collector:

1. Select the **Add a new Collector** option.
2. Select the platform for which you want to install the Sumo OpenTelemetry Collector.
```

> If you want to create a new Collector please select **Add a new Collector** option.

Reviewer: ```suggestion
```

> If you want to create a new Collector please select **Add a new Collector** option.

Reviewer 2: ```suggestion
If you want to create a new Collector select **Add a new Collector** option.
```

> Select the platform for which you want to install the Sumo OpenTelemetry Collector.

Reviewer: ```suggestion
```

This will generate a command which can be executed in the machine which needs to get monitored. Once executed it will install the Sumo Logic OpenTelemetry Collector agent.  

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Mac-OpenTelemetry/Mac-Collector.png')} alt="Collector" />

### Step 2: Configure integration

In this step we will be configuring the yaml required for macOS Collection.

There are no parameters required for the macOS app to install. One can add a custom tag if required in this step.

Click on the **Download YAML File** button to get the yaml file.

<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/Mac-OpenTelemetry/Mac-YAML.png')} alt="YAML" />

> :::note

Reviewer: This whole note section needs to be removed

Reviewer 3: All of Step 2?

By default the collector will be sending process metrics to Sumo Logic. Since the number of processes running can be very large, this may result in significant increase in Data Points per Minute (DPM) If you would like to narrow down the list of processes being monitored, this can be done by adding the following entry under the process section of the downloaded yaml.
```sh
process:
  include:
    names: [ <process name1>, <process name2> ... ]
    match_type: <strict|regexp>
```
:::

### Step 3: Sending metrics to Sumo

Once you have the yaml file downloaded in step 2, follow the below steps :

1.  Copy the yaml file to `/etc/otelcol-sumo/conf.d/` folder in the macOS instance which needs to be monitored.
2.  Restart the otelcol-sumo process using the below command 
```sh
 otelcol-sumo --config /etc/otelcol-sumo/sumologic.yaml --config "glob:/etc/otelcol-sumo/conf.d/*.yaml"
```

After successful execution of the above command, Sumo will start receiving the data from your host machine.

> Press **Next**. This will install the app in your Sumo Logic Org. The app consists of dashboards and monitors.

Reviewer: ```suggestion
Click **Next**. This will install the app in your Sumo Logic Org. The app consists of dashboards and monitors.
```

> Panels will start to fill automatically. It's important to note that each panel fills with data matching the time range query and received since the panel was created. Results won't immediately be available, but with a bit of time, you'll see full graphs and map.

Reviewer: ```suggestion
Panels will start to fill automatically. It's important to note that each panel fills with data matching the time range query and received since the panel was created. Results won't immediately be available, but within 20 minutes, you'll see full graphs and map.
```

## Sample Metric Query

Metric query from the panel - File System Utilization
```sh

> sumo.datasource=mac host=* device=* metric=system.filesystem.utilization | sum by host, device, type, mountpoint

Reviewer: ```suggestion
sumo.datasource=mac host.name=* device=* metric=system.filesystem.utilization | sum by host.name, device, type, mountpoint
```

```

## Sample Open Telemetry metric 

```sql

> {"queryId":"A","_source":"anema-mac-dc8d7942-8038-46f0-af33-9a3a06a73f72","state":"LAST_ACK","_sourceName":"anemamacsyslog","host":"anema-mac","os.type":"darwin","sumo.datasource":"mac","_sourceCategory":"anema/mac/sys/logs","metric":"system.network.connections","_collectorId":"000000000CAADE8D","_sourceId":"0000000000000000","unit":"{connections}","_sourceHost":"anema-mac","_collector":"anema-mac-dc8d7942-8038-46f0-af33-9a3a06a73f72","protocol":"tcp","max":0,"min":0,"avg":0,"sum":0,"latest":0,"count":14}

Reviewer: ```suggestion
{"queryId":"A","_source":"sumo-mac-dc8d7942-8038-46f0-af33-9a3a06a73f72","state":"LAST_ACK","_sourceName":"sumomacsyslog","host":"sumo-mac","os.type":"darwin","sumo.datasource":"mac","_sourceCategory":"sumo/mac/sys/logs","metric":"system.network.connections","_collectorId":"000000000CAADE8D","_sourceId":"0000000000000000","unit":"{connections}","_sourceHost":"sumo-mac","_collector":"sumo-mac-dc8d7942-8038-46f0-af33-9a3a06a73f72","protocol":"tcp","max":0,"min":0,"avg":0,"sum":0,"latest":0,"count":14}
```
